cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/Cluster.java

```diff
@@ -33,7 +34,8 @@
   public Cluster(String name, ClusterStatus status, Collection<Node> nodes, Map<String, String> properties) {
     this.name = name;
     this.status = status;
-    this.nodes = Collections.unmodifiableCollection(nodes);
+    this.nodes = nodes instanceof List ? Collections.unmodifiableList((List<Node>) nodes) :
```

Comment:
Why this change is needed? Also, shouldn't it better to do a defensive copy?

Reply:
It's because equals do not work with Collections.unmodifiableCollection :(. It affects testing, but also anyone else using equals. I can do a deep copy